pkl-core/src/main/java/org/pkl/core/http/HttpClient.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.http;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpTimeoutException;
+import java.nio.file.Path;
+import javax.net.ssl.SSLContext;
+
+/**
+ * An HTTP client.
+ *
+ * <p>To create a new HTTP client, use a {@linkplain #builder() builder}. To send {@linkplain
+ * HttpRequest requests} and retrieve their {@linkplain HttpResponse responses}, use {@link #send}.
+ * To release resources held by the client in a timely manner, use {@link #close}.
+ *
+ * <p>HTTP clients are thread-safe. Each client maintains its own connection pool and {@link
+ * SSLContext}. For efficiency reasons, clients should be reused whenever possible.
+ */
+public interface HttpClient extends AutoCloseable {
+
+  /** A builder of {@linkplain HttpClient HTTP clients}. */
+  interface Builder {
+    /**
+     * Sets the {@code User-Agent} header.
+     *
+     * <p>Defaults to {@code "Pkl/$version ($os $flavor)"}.
```

Comment:
> We intentionally kept this separate from pkl-core because on the Java side, we wanted our HTTP requests to defer to the host application's settings (e.g. trust the same CA roots).

Defaulting to the JDK's CA roots was my original goal, but I couldn't find a way to make it work. This is best discussed separately. 

In any case, if we want to support `HttpClient.Builder.addBuiltInCertificates` and `HttpClient.Builder.addDefaultCliCertificates`, which makes a lot of sense to me, then `IncludedCARoots.pem` must be accessible from `pkl-core`. As a consequence, it cannot reside in `pkl-commons` or `pkl-commons-cli` (its original location) because neither is a dependency of `pkl-core`.

Reply:
> I don't see any downsides of having this file in pkl-core.

The main downside is filesize; the file is 213Kb. All of our JVM users end up with this file in their classpath and likely none of them will need it.

Reply:
How about putting the built-in certs into a `pkl-certs` JAR? Then everyone who wants them can have them, without pulling in any other JAR. (This isn't even limited to JVM use cases because a JAR is just a ZIP.) And everyone else can stay away from them. And we can stop guessing.

Reply:
That sounds fine with me!

Reply:
Should this JAR be versioned with Pkl? I'm asking because it will rarely change. And should it actually be a ZIP file, i.e., have a `.zip` extension? `ClassLoader.getResource(pathToZipFile)` is officially supported.

Reply:
> <p>Defaults to {@code "Pkl/$version ($os; $flavor)"}.
fixed

Reply:
> Should this JAR be versioned with Pkl? I'm asking because it will rarely change. And should it actually be a ZIP file, i.e., have a .zip extension? ClassLoader.getResource(pathToZipFile) is officially supported.

I was thinking it would be its own project (pkl-certs, sibling of pkl-cli, etc). And it would publish its own jar to Maven. So yeah, versioned with Pkl.

What were you thinking here?

Reply:
For now I did `pkl-certs` because this was easiest. But this *could* be a separate GitHub project with initial version 1. And it *could* publish a `.zip` file to Maven Central. I don't really mind but wanted to mention the alternatives.

Reply:
Ah I see. Let's keep it here.